### PR TITLE
Remove redundant hasErrors variables from linters

### PR DESCRIPTION
**Pull request by vinyldarkscratch**
This PR removes the redundant hasErrors variable from the linter scripts, as we now rely on the length of errors to indicate whether we have any.

### CODE REVIEW

File: test/linter/test-versions.js [hunk 1]
```diff
@@ -85,15 +81,13 @@ function checkVersions(supportData, relPath, logger) {
               )
             ) {
               logger.error(chalk`{red → {bold ${relPath}} - {bold version_removed: "${statement.version_removed}"} must be greater than {bold version_added: "${statement.version_added}"}}`);
-              hasErrors = true;
             }
           }
         }
       }
     }
   }
```
Comment (ExE-Boss) [inline test/linter/test-versions.js]:
```suggestion
```

**Comment by ExE-Boss** [comment]
#4813

**Comment by Elchi3** [comment]
Do we also need to remove `hasErrors` from `test-links.js`?

**Comment by Elchi3** [comment]
Ah, forget what I was saying! I see you are doing that in https://github.com/mdn/browser-compat-data/pull/5187.

**Pull request merged**